### PR TITLE
Benign training fix

### PR DESCRIPTION
**Pull request by anindya-paul**
My findings on difference between benign and DP training: I was experimenting on the Opacus [“wrap_data_loader”](https://github.com/pytorch/opacus/blob/5cc574ff877b0be5634dde8fdd5130b7090491a6/opacus/utils/batch_memory_manager.py#L80) (“memory_safe_data_loader” in our code) in our training pipeline (“opacus_integation” branch)  and after outputting the data loader output what I see is it is using “UniformWithReplacementSampler” as a BatchSpllittingSampler which is not correct for a benign batch sampler. There are a few other changes but I think this is impacting the most on the benign case.

Based on my observation, I made just one change: the “[BatchMemoryManager](https://github.com/pytorch/opacus/blob/5cc574ff877b0be5634dde8fdd5130b7090491a6/opacus/utils/batch_memory_manager.py#L114)” class which generates the data loader wrapper to be used only during DP training. For benign training, the original “train_dataloader” with the same “batch_size” mentioned in the config_*.yaml will be used and hence standard batch_sampler will be used for benign. I tested using MNIST benign training and from the logs generated using CBICA GaNDLF Master (Torch 1.8.2) and opacus_integration_benigntrainingfix  (Torch 1.11.0), I see now they match quite well!

##########################################

Fixes #ISSUE_NUMBER

## Proposed Changes

- 

## Checklist

- [ ] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [ ] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [ ] Non-breaking change (would not break existing functionality): please provide as many details as possible for any breaking change 
- [ ] Function/class source code documentation added/updated
- [ ] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [ ] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/version.py)
- [ ] If adding a submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [ ] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [ ] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [ ] Tests added or modified to [cover the changes](https://app.codecov.io/gh/CBICA/GaNDLF); if coverage is reduced, please give explanation

### CODE REVIEW

**Review by brandon-edwards — COMMENTED** [review]
LGTM (I agree with Sarthak's suggestions)

File: GANDLF/compute/training_loop.py [hunk 1]
```diff
@@ -82,131 +82,139 @@ def train_network(model, train_dataloader, optimizer, params):
...
+        memory_safe_data_loader = train_dataloader
 
-        for batch_idx, (subject) in enumerate(
-            tqdm(memory_safe_data_loader, desc="Looping over training data")
-        ):
+    # with BatchMemoryManager(
+    #     data_loader=train_dataloader,
+    #     max_physical_batch_size=MAX_PHYSICAL_BATCH_SIZE,
+    #     optimizer=optimizer,
+    # ) as memory_safe_data_loader:
```
[the remark this answers is not in the document]
Comment (anindya-paul) [reply GANDLF/compute/training_loop.py:103]: Done

File: GANDLF/compute/training_loop.py [hunk 2]
```diff
@@ -82,131 +82,139 @@ def train_network(model, train_dataloader, optimizer, params):
 
     # in the case of opacus DP training, this will dynamically change over batch iterations
     batch_size = params["batch_size"]
-    MAX_PHYSICAL_BATCH_SIZE = batch_size
+    # MAX_PHYSICAL_BATCH_SIZE = batch_size
```
[the remark this answers is not in the document]
Comment (anindya-paul) [reply GANDLF/compute/training_loop.py:85]: Done

**Comment by anindya-paul** [comment]
Thanks @sarthakpati and @sarthakpati. I ran `black .` as suggested and now all checks have passed 👍

**Pull request merged**